### ISSUE TITLE
Problem with Namespace Tests

### ISSUE THREAD

**Issue by dietzc** [imagej/imagej-ops#153]
Hi @ctrueden,

while testing the `GlobalNamespace` in the branch https://github.com/imagej/imagej-ops/tree/gauss-refactor, I received the following error messages:

(a)
```
[ERROR] Mismatched inputs:
	op = net.imagej.ops.gauss.DefaultGaussRAI
	method = public abstract net.imglib2.RandomAccessibleInterval net.imagej.ops.OpService.gauss(net.imglib2.RandomAccessibleInterval,double[],net.imglib2.outofbounds.OutOfBoundsFactory)
```
(b)
```
[ERROR] Last argument should be varargs for method:
	public abstract net.imglib2.RandomAccessibleInterval net.imagej.ops.OpService.gauss(net.imglib2.RandomAccessibleInterval,net.imglib2.RandomAccessibleInterval,double[])
```
(a) & (b) ;-)
```
[ERROR] Last argument should be varargs for method:
	public abstract net.imglib2.RandomAccessibleInterval net.imagej.ops.OpService.gauss(net.imglib2.RandomAccessibleInterval,double[])
```

It (a) seems that the optional output of `DefaultGaussRAI` needs to be present in the method signature and (b) that optional parameters at the end of the signature, have to be present, too. right? If so, why?

Thanks,
Christian

**Comment by ctrueden** [comment]
The `(b)` error means you should write `double... sigmas` instead of `double[] sigmas` for the signature whose last argument is the `sigmas` parameter. This is for maximum convenience, because there is no real downside to using varargs whenever possible.

**Comment by ctrueden** [comment]
Regarding whether optional arguments need to be present: actually, all method signatures (with and without optional arguments) must be present.

**Comment by dietzc** [comment]
fixed (b). Thanks!

**Comment by ctrueden** [comment]
Let me know if you need more help with `(a)` or if you think there is a bug in the testing.

**Comment by dietzc** [comment]
Error is: 
```
[ERROR] Mismatched inputs:
	op = net.imagej.ops.gauss.DefaultGaussRAI
	method = public abstract net.imglib2.RandomAccessibleInterval net.imagej.ops.OpService.gauss(net.imglib2.RandomAccessibleInterval,double[],net.imglib2.outofbounds.OutOfBoundsFactory)
```
If I understand the error message correctly, then it says, that I've added a signature which is actually not matching any possible parameter combinations of `DefaultGaussRAI`. However, the this signature certainly exists in `DefaultGaussRAI`, as it has an optional output.

**Comment by ctrueden** [comment]
I checked out the branch. In addition to the method signature issue, there are problems with `gauss` methods delegating to one another, which produce a `GaussTest.test » StackOverflow` test failure. These methods should not delegate to one another, but rather call `OpService.run` directly in every case. At least, that's how we did _everything_ else so far, so for consistency I'd like to do that here too. My intuition is that there are potentially subtle behavior problems if you do otherwise, because which op is coded to run might differ across the signatures. In particular: `gauss(Object...)` uses `Ops.Gauss.NAME` which includes every `gauss` op, whereas the built-ins are supposed to specify which concrete implementation directly whenever possible.

**Comment by ctrueden** [comment]
Anyway, I'm fixing things up, and will push soon.

**Comment by ctrueden** [comment]
You forgot the `@OpMethod` annotation on some of the methods in `OpService` interface.

**Comment by ctrueden** [comment]
Or... these signatures don't even correspond to any ops directly? OK, I'm going to shut up and figure this out now.

**Comment by dietzc** [comment]
I left them away on purpose, as these are just helpers... Maybe thats the wrong way to do it..

**Comment by ctrueden** [comment]
OK, fixed in 7e70a4d51383b7df528c1f3c24ea83f01846d8eb. See also 2d7652b92192cc211073572594387967f33daa0e. Feel free to squash these however you see fit.
